src/main/java/dev/j3fftw/litexpansion/machine/MetalForge.java

```diff
@@ -111,15 +111,11 @@ private void craft(Player p, Block b, Inventory inv, ItemStack[] recipe, ItemSta
         diamondBlock.setType(Material.AIR);
     }
 
-    private boolean canCraft(Inventory inv, List<ItemStack[]> inputs, int i) {
-        for (ItemStack converting : inputs.get(i)) {
-            if (converting != null) {
-                for (int j = 0; j < inv.getContents().length; j++) {
-                    if (j == (inv.getContents().length - 1)
-                        && !SlimefunUtils.isItemSimilar(converting,
-                        inv.getContents()[j], true)) {
-                        return false;
-                    } else if (SlimefunUtils.isItemSimilar(inv.getContents()[j], converting, true)) break;
+    private boolean canCraft(Inventory inv, ItemStack[] recipe) {
+        for (int j = 0; j < inv.getContents().length; j++) {
+            if (!SlimefunUtils.isItemSimilar(inv.getContents()[j], recipe[j], true)) {
+                if (!SlimefunUtils.isItemSimilar(inv.getContents()[j], recipe[j], false)) {
```

Comment:
what in the fuck is this double check?? checking the lore is useless to begin with since all our items have IDs

but still, what in the fuck is this